src/middlewared/middlewared/plugins/smart.py

```diff
@@ -296,6 +297,101 @@ class Config:
 
         return response
 
+    @accepts(
+        List(
+            'disks', items=[
+                Dict(
+                    'disk_run',
+                    Str('identifier', required=True),
+                    Str('mode', enum=['FOREGROUND', 'BACKGROUND'], default='BACKGROUND'),
+                    Str('type', enum=['LONG', 'SHORT', 'CONVEYANCE', 'OFFLINE'], required=True),
+                )
+            ]
+        )
+    )
+    async def manual_test(self, disks):
+        """
+        Run manual SMART tests for `disks`.
+
+        `type` indicates what type of SMART test will be ran and must be specified.
+        """
+        verrors = ValidationErrors()
+        if not disks:
+            verrors.add(
+                'manual_test.disks',
+                'Please specify at least one disk.'
+            )
+        else:
+            test_disks_list = []
+            disks_data = await self.middleware.call('disk.query')
+            devices = await camcontrol_list()
+
+            for disk in disks:
+                for d in disks_data:
+                    if disk['identifier'] == d['identifier']:
+                        current_disk = d
+                        test_disks_list.append({
+                            'disk': current_disk['name'],
+                            **disk
+                        })
+                        break
+                else:
+                    verrors.add(
+                        'manual_test.disks.identifier',
+                        f'{disk["identifier"]} is not valid. Please provide a valid disk identifier.'
+                    )
+                    continue
+
+                if current_disk['name'] is None or current_disk['name'].startswith('nvd'):
```

Comment:
Whats the story about nvd?

Reply:
I am not sure, this condition exists in `annotate_disk_smart_tests` and my usage for running smartctl tests is almost the same so i copied it over keeping the usage consistent. I was thinking maybe `camcontrol_list` function has something to do with it and is unable to retrieve disks starting with that ?
Anyways, the idea was it would be brought up in the PR and if it's out of scope, we can have it removed.
@themylogin can you please confirm if this check is requried ?

Reply:
Yeah well, there isnt a way to map nvd to nvme devices yet, so I believe thats why.

But lets make the error message more clear if thats really the case. NVMe users will be clueless.

Reply:
Perfect, i'll have it updated